### DISCUSSION TITLE
Visual studio and TypeScript version for CommerceSDK

### DISCUSSION THREAD

**Discussion by RethaLam** [microsoft/Dynamics365Commerce.Solutions#28] [Q&A] [answered]
I have started with the switch over from RetailSDK to CommerceSDK. We are an in-house development team, not ISV or Microsoft partner, so I don't need to keep RetailSDK around. I plan on copying everything over to CommerceSDK so that the users can test it while we can continue with customization on RetailSDK. Work will be duplicated at the beginning but that is ok. I rather have the 2 separate.

At one point we had to upgrade Visual Studio from 2015 to 2017 for RetailSDK. Even though VS 2019 was available, retailSDK couldn't function on it. The typeScript version prerequisite was also v2.2.2 for RetailSDK.
What are the prerequisites for CommerceSDK regarding Visual Studio and TypeScript? I know it doesn't work with TypeScript 2.2.2 because the only line now in the tsconfig-file is the extend from the pos-tsconfig-base.json line. When building the empty POSExtension project, it generates a strictFunctonType error on TypeScript 2.2.2. 
https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/pos-extension/create-pos-extension-package just gives a tip about v4.2.3 for TypeScript as an example. 
Do I need to upgrade my visual Studio to 2019 because currently VS only has 3.1 for TypeScript. I can download the latest TypeScript from GitHub, v4.9.4, but it installs only as a stand-alone. So how do I get it into Tools for Visual Studio. Or will upgrading to VS 2019 gives me the version I need for CommerceSDK? I also read that VS 2022 dropped TypeScript under the Tools and it will only be available as a nuget package. 
Please note that I prefer to work with Visual Studio and not Visual Studio code. So I need to get this working for Visual Studio. In the meantime, to get the empty POSExtension project to build successfully, I installed via nuget Manager Microsoft.TypeScript.MSBuild with version 4.23, but Tools for Visual Studio still has v2.2.2 and v3.1 checked for typeScript. The 2 needs to have the same version.  

Any help is much appreciated.

**Comment by madyke** [comment, marked as answer]
Hi @RethaLam - for Commerce SDK and Store Commerce, Visual Studio 2022, 2019, and 2017 should all be sufficient.

For the TypeScript version, we recommend TypeScript 4+ when working on Commerce SDK. I'm not familiar which version of TS each version of Visual Studio comes with by default, but there is some documentation available for setting up TypeScript as a part of the POS extension projects that may help (see step 7 of [Create a POS extension package project](https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/pos-extension/create-pos-extension-package)). [TypeScript's website](https://www.typescriptlang.org/download) also has some info on how you can get TypeScript configured with Visual Studio.

Otherwise, if you're having issues with support for older versions of Visual Studio, then if it's an option for you, upgrading to a newer version of Visual Studio such as 2022 should provide you better support out of the box and the latest features.
